client/state/login/oauth2/reducer.js

```diff
@@ -71,11 +73,29 @@ export const initialClientsData = {
 	},
 };
 
+const getClientIdFromSignupUrl = ( signupUrl ) => {
+	const urlParts = signupUrl.split( '?' );
+	const queryString = urlParts[ 1 ];
+
+	const params = qs.parse( queryString );
+	const oauth2Redirect = params.oauth2_redirect;
+	const oauth2RedirectParts = oauth2Redirect.split( '?' );
+	const oauth2RedirectQueryString = oauth2RedirectParts[ 1 ];
+	const oauth2RedirectParams = qs.parse( oauth2RedirectQueryString );
+
+	return oauth2RedirectParams.client_id;
```

Comment:
What if for some reasons one parameter is missing from the signup url? Right now we're assuming the url is always correctly formatted, but should we take a more defensive approach here?